File: Codelearn/Sasuke 26/PowerOfThreeNotNine.js
function powerOfThreeNotNine(m, n){
    let nine = [];
    for(let i = 0; i <= 10; i++)
    {
        let pow = Math.pow(9,i);
        if(pow >= m)
        {
            if(pow <= n) nine.push(pow);
            else break;
        }
    }
    let count = 0;
    for(let i = 0; i <= 20; i++)
    {
        let pow = Math.pow(3,i);
        if(pow >= m)
        {
            if(pow <= n)
            {
                if(!nine.includes(pow)) count++;
            }else break;
        }
    }
    return count;
}
 

File: Codelearn/Traning/PowerOfThreeNotNine.java
int powerOfThreeNotNine(int m, int n)
{
    ArrayList<Integer> nine = new ArrayList<Integer>();
    for(int i = 0; i <= 10; i++)
    {
        int pow = (int) Math.pow(9,i);
        if(pow >= m)
        {
            if(pow <= n) nine.add(pow);
            else break;
        }
    }
    int count = 0;
    for(int i = 0; i <= 20; i++)
    {
        int pow = (int) Math.pow(3,i);
        if(pow >= m && pow <= n && !nine.contains(pow)) count++;
    }
    return count;
}
 